src/components/picker/PickerSelectionControlBar.tsx

```diff
@@ -0,0 +1,72 @@
+import React, {useMemo, useCallback} from 'react';
+import Button from '../button';
+import Checkbox from '../checkbox';
+import View from '../view';
+import Text from '../text';
+import {PickerItemProps, PickerProps, PickerContextProps, PickerMultiValue} from './types';
+
+export type PickerSelectionControlBarProps = PickerProps['selectionStatus'] &
+  Partial<PickerContextProps> & {
+    value?: PickerMultiValue;
+    items?: PickerItemProps[];
+  };
+
+export default function PickerSelectionControlBar(props: PickerSelectionControlBarProps) {
+  const {
+    selectAllType = 'none',
+    buttonProps = {},
+    checkboxProps = {},
+    containerStyle,
+    value = [],
+    items = [],
+    toggleAllItemsSelection,
+    showLabel = true,
+    customLabel
+  } = props;
+
+  const availableItems = useMemo(() => items.filter(item => !item.disabled).map(item => item.value), [items]);
+  const isAllSelected = value.length === availableItems.length;
+  const checkboxIndeterminate = value.length > 0 && value.length < availableItems.length;
+  const defaultLabel = `${value.length} Selected ${isAllSelected ? '(All)' : ''}`;
+
+  const handlePress = useCallback(() => {
+    const newSelectionState = !isAllSelected;
+    toggleAllItemsSelection?.(availableItems, newSelectionState);
```

Comment:
This API is a little weird, you should not need to pass the items IMO.
Generally I'd prefer the logic for selection will be in `usePickerSelection` if possible.